### ISSUE TITLE
Plugin names with parentheses fails to build on macOS

### ISSUE THREAD

**Issue by directmusic** [free-audio/clap-juce-extensions#120]
I have a project where the PRODUCT_NAME in the CMake file is named "Transition Filter (No GUI)"

The Audio Unit and VST3 build perfectly fine on macOS, but the CLAP build fails with this error:
```
/bin/sh: -c: line 0: syntax error near unexpected token `('
/bin/sh: -c: line 0: `/opt/homebrew/Cellar/cmake/3.25.0/bin/cmake -E echo Installing\ /Users/joe/Documents/GitHub/TransitionFilter/build-macos-no-ui/TransitionFilter_artefacts/Debug/CLAP/Transition\ Filter\ (No\ GUI).clap\ to\ ~/Library/Audio/Plug-Ins/CLAP/'
make[3]: *** [TransitionFilter_artefacts/Debug/CLAP/Transition Filter (No GUI).clap/Contents/MacOS/Transition Filter (No GUI)] Error 2
make[3]: *** Archive member `TransitionFilter_artefacts/Debug/CLAP/Transition Filter (No GUI).clap/Contents/MacOS/Transition Filter (No GUI)' may be bogus; not deleted
```

I have tried escaping the parentheses in the PRODUCT_NAME in the JUCE CMake file like this "Transition Filter \\(No GUI\\)" but to no avail.

**Comment by baconpaul** [comment]
oh darnit this is the copy plugin step failing.
and just on an echo too!

Can you remove this (and only this) line and see if it fixes it?

https://github.com/free-audio/clap-juce-extensions/blob/cf93cac7ca6a1c0c548c2bda85d49e9bfdb979e1/cmake/ClapTargetHelpers.cmake#L155

**Comment by directmusic** [comment]
Oh, my apologies. I accidentally copied an error from when I was already messing around with adding `VERBATIM` to those commands which I was hoping could be a solution.

```
/bin/sh: -c: line 0: syntax error near unexpected token `('
/bin/sh: -c: line 0: `/opt/homebrew/Cellar/cmake/3.25.0/bin/cmake -E copy_if_different /Users/joe/Documents/GitHub/TransitionFilter/clap-juce-extensions/cmake/macos_bundle/PkgInfo /Users/joe/Documents/GitHub/TransitionFilter/build-macos-no-ui/TransitionFilter_artefacts/Debug/CLAP/Transition\ Filter\ (No\ GUI).clap/Contents/MacOS/..'
make[3]: *** [TransitionFilter_artefacts/Debug/CLAP/Transition Filter (No GUI).clap/Contents/MacOS/Transition Filter (No GUI)] Error 2
make[3]: *** Archive member `TransitionFilter_artefacts/Debug/CLAP/Transition Filter (No GUI).clap/Contents/MacOS/Transition Filter (No GUI)' may be bogus; not deleted
```

This is where the error first occurs.

https://github.com/free-audio/clap-juce-extensions/blob/cf93cac7ca6a1c0c548c2bda85d49e9bfdb979e1/cmake/ClapTargetHelpers.cmake#L105-L107

Adding VERBATIM to these lines as well as as the following lines appears to silence that error but the build still fails.

https://github.com/free-audio/clap-juce-extensions/blob/cf93cac7ca6a1c0c548c2bda85d49e9bfdb979e1/cmake/ClapTargetHelpers.cmake#L155-L157

Here is the error after my attempt to fix:

```
Installing /Users/joe/Documents/GitHub/TransitionFilter/build-macos-no-ui/TransitionFilter_artefacts/Debug/CLAP/Transition Filter (No GUI).clap to ~/Library/Audio/Plug-Ins/CLAP/
Error copying directory from "/Users/joe/Documents/GitHub/TransitionFilter/build-macos-no-ui/TransitionFilter_artefacts/Debug/CLAP/Transition Filter (No GUI).clap" to "~/Library/Audio/Plug-Ins/CLAP/Transition Filter (No GUI).clap".
make[3]: *** [TransitionFilter_artefacts/Debug/CLAP/Transition Filter (No GUI).clap/Contents/MacOS/Transition Filter (No GUI)] Error 1
make[3]: *** Archive member `TransitionFilter_artefacts/Debug/CLAP/Transition Filter (No GUI).clap/Contents/MacOS/Transition Filter (No GUI)' may be bogus; not deleted
make[2]: *** [CMakeFiles/TransitionFilter_CLAP.dir/all] Error 2
make[1]: *** [CMakeFiles/TransitionFilter_CLAP.dir/rule] Error 2
make: *** [TransitionFilter_CLAP] Error 2
```

**Comment by baconpaul** [comment]
OK I'll take a look in the morning with a test project then. If you find anything in the interim let me know! Thanks!

**Comment by baconpaul** [comment]
Ahh yeah I'm able to reproduce this in the example project.  Should be able to peek tomorrow

**Comment by baconpaul** [comment]
Ahh I think I found it. It at least works in my build test with the GainPlugin example. (I have not run it but the build succeeds).

```
diff --git a/cmake/ClapTargetHelpers.cmake b/cmake/ClapTargetHelpers.cmake
index 543e841..bf78c0f 100644
--- a/cmake/ClapTargetHelpers.cmake
+++ b/cmake/ClapTargetHelpers.cmake
@@ -105,6 +105,7 @@ function(clap_juce_extensions_plugin_internal)
                COMMAND ${CMAKE_COMMAND} -E copy_if_different "${cjd}/cmake/macos_bundle/PkgInfo" "$<TARGET_FILE_DIR:${claptarget}>/.."
                COMMAND ${CMAKE_COMMAND} -E make_directory "$<TARGET_FILE_DIR:${claptarget}>/../Resources"
                COMMAND ${CMAKE_COMMAND} -E copy_if_different "${cjd}/cmake/macos_bundle/clap.icns" "$<TARGET_FILE_DIR:${claptarget}>/../Resources"
+               VERBATIM
                )
     else()
         set_target_properties(${claptarget} PROPERTIES
```

Can you apply this diff? namely add the text "VERBATIM" at the end of the command include the copy_if_different. That should match the other targets.

If that works I'd be happy to merge that change also.

Thanks for your patience.

**Comment by directmusic** [comment]
That does fix it if `COPY_PLUGIN_AFTER_BUILD` is set to `FALSE`

However if it set it to `TRUE` in my CMakeLists file it fails later with a similar error. This also occurs on Linux as well.

Here is a [test repo demonstrating the bug.](https://github.com/directmusic/juce_clap_bug_test) There is a build.sh file in there which shows the commands I am using to test.

I added VERBATIM to the section which implements the COPY_PLUGIN_AFTER_BUILD option to address the next errors however it then failed when trying to expand the `~/` I did find the fix for that portion and just submitted a PR that addresses it on macOS and Linux. I am happy to give you authorship of this change however if you'd prefer.
